lib/buffer.js

```diff
@@ -999,15 +997,17 @@ function adjustOffset(offset, length) {
   // Use Math.trunc() to convert offset to an integer value that can be larger
   // than an Int32. Hence, don't use offset | 0 or similar techniques.
   offset = Math.trunc(offset);
-  // `x !== x`-style conditionals are a faster form of `isNaN(x)`
-  if (offset === 0 || offset !== offset) {
+  if (offset === 0) {
     return 0;
-  } else if (offset < 0) {
+  }
+  if (offset < 0) {
     offset += length;
     return offset > 0 ? offset : 0;
-  } else {
-    return offset < length ? offset : length;
   }
+  if (Number.isNaN(offset)) {
```

Comment:
Any reason for not merging this with the `offset === 0` case on line 1000?

Reply:
`NaN` is just a safety check. The other cases are not. Therefore I prioritized the other case. Due to your comment I actually saw that it is possible to move the NaN check down to be the last.